boards/shields/wiznet_wiz550io/wiznet_wiz550io.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2024 Grant Ramsay
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This driver provides the initialsation sequence required for
+ * WIZ550io to configure the W5500 Ethernet controller with its
+ * embedded unique MAC address.
+ */
+
+#define DT_DRV_COMPAT wiznet_wiz550io
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(eth_w5500, CONFIG_ETHERNET_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/net/ethernet.h>
+
+#include "eth.h"
+#include "eth_w5500_priv.h"
+
+struct wiz550io_config {
+	const struct device *w5500_dev;
+	struct gpio_dt_spec reset;
+	struct gpio_dt_spec ready;
+	bool use_mac_addr;
+};
+
+/* WIZ550io documentation recommends 150ms delay after HW reset
+ * for PIC12F519 MCU to configure the W5500.
+ */
+#define WIZ550IO_RESET_DELAY K_MSEC(150)
+
+static int wiz550io_init(const struct wiz550io_config *config)
+{
+	int err;
+	const struct w5500_config *w5500_config = config->w5500_dev->config;
+	struct w5500_runtime *w5500_runtime = config->w5500_dev->data;
+
+	if (!spi_is_ready_dt(&w5500_config->spi)) {
```

Comment:
the spi could be figured out from the DT in this file instead of looking at the ethernet mac config